### PR TITLE
adhocquery v-1.0

### PR DESCRIPTION
**Pull request by iamrupok**
PR-ADHOCQUERY

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/api/AdHocApiResource.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        this.context.authenticatedUser();
+
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+
+        final AdHocData adhoc = this.adHocReadPlatformService.retrieveOne(adHocId);
+
+        return this.toApiJsonSerializer.serialize(settings, adhoc, this.RESPONSE_DATA_PARAMETERS);
+    }
+
+/*    *//**
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/adhocquery/api/AdHocApiResource.java:131]: Please remove unused/commented code

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocWritePlatformServiceJpaRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    @Transactional
+    @Override
+    public CommandProcessingResult updateAdHocQuery(final Long roleId, final JsonCommand command) {
+        try {
+            this.context.authenticatedUser();
+
+            this.adHocCommandFromApiJsonDeserializer.validateForUpdate(command.json());
+
+            final AdHoc adHoc = this.adHocRepository.findOne(roleId);
+            if (adHoc == null) { throw new RoleNotFoundException(roleId); }
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocWritePlatformServiceJpaRepositoryImpl.java:113]: if adHoc is  null you should throw AdHocNotFoundException  not RoleNotFoundException?

File: fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java [hunk 3]
```diff
@@ -2904,4 +2904,43 @@ public CommandWrapperBuilder deleteSmsCampaign(final Long resourceId) {
         this.href = "/smscampaigns/"+resourceId;
         return this;
     }
+	public CommandWrapperBuilder disableAdHoc(Long adHocId) {
+        this.actionName = "DISABLE";
+        this.entityName = "ADHOC";
+        this.entityId = adHocId;
+        this.href = "/adhoc/" + adHocId + "/disbales";
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/commands/service/CommandWrapperBuilder.java:2911]: make it disable. I believe it should be command parameter?

File: fineract-provider/src/main/resources/sql/migrations/core_db/V323__adhocquery.sql [hunk 4]
```diff
@@ -0,0 +1,48 @@
+--
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V323__adhocquery.sql:1]: Current migration number 327. So Your migration script number should 328

**Comment by iamrupok** [comment]
created a new pull request
#365

**Comment by nazeer1100126** [comment]
@iamrupok close this PR

**Pull request closed without merging**